Utilities/rANS/include/rANS/DedupEncoder.h

```diff
@@ -104,31 +94,16 @@ const stream_IT DedupEncoder<coder_T, stream_T, source_T>::process(const stream_
       duplicates.emplace(std::distance(inputBegin, symbolIter) - 1, numDuplicates);
     }
 
-    return std::tuple(++dedupIT, coder.putSymbol(outputIter, encoderSymbol, this->mProbabilityBits));
+    return std::tuple(++dedupIT, coder.putSymbol(outputIter, encoderSymbol));
```

Comment:
I would recommend using a `std::pair` if you are returning a tuple of two, especially if that saves you from including `<tuple>`. Even better, have a struct of two elements so you do not have to remember who is first / second.

Reply:
Actually I didn't know `std::pair` and structs supported structured bindings in C++17. Thanks for the hint :)